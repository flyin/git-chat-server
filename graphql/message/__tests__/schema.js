const _ = require('lodash');
const mongoose = require('mongoose');
const { graphql } = require('graphql');

const Channel = require('../../channel');
const Message = require('../../message');
const { clearDatabase } = require('../../../commands/clear-database');
const schema = require('../../schema');

describe('Channel schema test case', () => {
  beforeEach(async () => await clearDatabase());
  afterAll(async () => await mongoose.disconnect());

  it('Should create message', async () => {
    const query = `
      mutation createMessage($text: String!, $channelId: String!) {
        createMessage(text: $text, channelId: $channelId) {
          text
          channel {
            name
          }
        }
      }
    `;

    const channel = await Channel.handlers.createChannel({ name: 'some-channel' });
    const result = await graphql(schema, query, {}, {}, { channelId: channel._id, text: 'Some message' });
    expect(result.data).toMatchSnapshot();
  });

  it('Should return channel messages', async () => {
    const query = `
      query messages($channelId: ID!) {
        messages(channelId: $channelId) {
          channel {
            name
          }
          text
        }
      }
    `;

    const channel = await Channel.handlers.createChannel({ name: 'some-channel' });
    const anotherChannel = await Channel.handlers.createChannel({ name: 'some-another-channel' });

    const messages = [
      { channelId: channel._id, text: 'message-1' },
      { channelId: channel._id, text: 'message-2' },
      { channelId: channel._id, text: 'message-3' },
      { channelId: channel._id, text: 'message-4' }
    ];

    const anotherMessages = [
      { channelId: anotherChannel._id, text: 'message-10' },
      { channelId: anotherChannel._id, text: 'message-11' },
      { channelId: anotherChannel._id, text: 'message-12' }
    ];

    await Promise.all([...messages, ...anotherMessages].map(({ channelId, text }) => Message.handlers.createMessage({ channelId, text })));
    const result = await graphql(schema, query, {}, {}, { channelId: channel._id });
    expect(result.data.messages.length).toBe(messages.length);

    result.data.messages.forEach(message => {
      expect(_.findIndex(messages, {text: message.text})).toBeGreaterThanOrEqual(0);
      expect(message).toMatchSnapshot(`message-${message.text}`);
    });
  });
});
