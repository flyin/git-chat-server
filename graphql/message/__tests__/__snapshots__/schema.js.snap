// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel schema test case Should create message 1`] = `
Object {
  "createMessage": Object {
    "channel": Object {
      "name": "some-channel",
    },
    "text": "Some message",
  },
}
`;

exports[`message-message-1 1`] = `
Object {
  "channel": Object {
    "name": "some-channel",
  },
  "text": "message-1",
}
`;

exports[`message-message-2 1`] = `
Object {
  "channel": Object {
    "name": "some-channel",
  },
  "text": "message-2",
}
`;

exports[`message-message-3 1`] = `
Object {
  "channel": Object {
    "name": "some-channel",
  },
  "text": "message-3",
}
`;

exports[`message-message-4 1`] = `
Object {
  "channel": Object {
    "name": "some-channel",
  },
  "text": "message-4",
}
`;
